src/topology/topological_fiber_bundle.lean

```diff
@@ -979,3 +1051,111 @@ begin
 end
 
 end topological_fiber_bundle_core
+
+section topological_fiber_prebundle
+
+variables (F) {Z : Type*} [topological_space B] [topological_space F] {proj : Z → B}
+
+/-- This structure permits to define a fiber bundle when trivializations are given as local
+equivalences but there is not yet a topology on the total space. The total space is hence given a
+topology in such a way that there is a fiber bundle structure for which the local equivalences
+are also local homeomorphism and hence local trivializations. -/
+@[nolint has_inhabited_instance]
+structure topological_fiber_prebundle (proj : Z → B) :=
+(trivialization_at : B → prebundle_trivialization F proj)
+(trivialization_at_mem : ∀ x : B, x ∈ (trivialization_at x).base_set)
+(cont_triv_inter : ∀ x y : B, continuous_on ((trivialization_at x) ∘
+  (trivialization_at y).to_local_equiv.symm) ((trivialization_at y).target ∩
+  ((trivialization_at y).to_local_equiv.symm ⁻¹' (trivialization_at x).source)))
+
+variable {F}
+
+/-- Topology on the total space that will make the prebundle into a bundle. -/
+def topological_fiber_prebundle.total_space_topology (a : topological_fiber_prebundle F proj) :
+  topological_space Z :=
+⨆ x : B, coinduced (a.trivialization_at x).set_symm (subtype.topological_space)
+
+lemma preimage_nhds_within_coinduced {α : Type*} {β : Type*} [topological_space α] {π : α → β}
+  {s : set β} {t : set α} {a : α} (h : a ∈ t) (ht : is_open t)
+  (hs : s ∈ @nhds β (topological_space.coinduced (λ x : t, π x) subtype.topological_space) (π a)) :
+  π ⁻¹' s ∈ 𝓝[t] a :=
+begin
+  letI := topological_space.coinduced (λ x : t, π x) subtype.topological_space,
+  rcases mem_nhds_iff.mp hs with ⟨V, hVs, V_op, mem_V⟩,
+  refine mem_nhds_within_iff_exists_mem_nhds_inter.mpr ⟨π ⁻¹' V, mem_nhds_iff.mpr ⟨t ∩ π ⁻¹' V,
+    inter_subset_right t (π ⁻¹' V), _, mem_sep h mem_V⟩, subset.trans (inter_subset_left _ _)
+    (preimage_mono hVs)⟩,
+  obtain ⟨u, hu1, hu2⟩ := is_open_induced_iff.mp (is_open_coinduced.1 V_op),
+  rw [preimage_comp] at hu2,
+  rw [set.inter_comm, ←(subtype.preimage_coe_eq_preimage_coe_iff.mp hu2)],
+  exact hu1.inter ht,
+end
+
+lemma topological_fiber_prebundle.continuous_inv_triv_at (a : topological_fiber_prebundle F proj)
+  (x : B) : @continuous_on _ _ _ a.total_space_topology (a.trivialization_at x).to_local_equiv.symm
+  (a.trivialization_at x).target :=
+begin
+  refine id (λ z H, id (λ U h, preimage_nhds_within_coinduced H (a.trivialization_at x).open_target
+  (le_def.1 (nhds_mono _) U h))),
+  exact le_supr _ x,
+end
+
+lemma subtype.preimage_val_eq_preimage_val_iff' {α : Type*} (s t u : set α) :
```

Comment:
this lemma has nothing to do here. Why not put it close to `subtype.preimage_val_eq_preimage_val_iff`? (and also rename it to `subtype.preimage_coe_eq_preimage_coe_iff`)

Reply:
Yep it was meant to be moved but I forgot it there...